File: 2021/day01/part01/main.zig
const std = @import("std");
const input = @embedFile("input.txt");
// const input = @embedFile("example.txt");

pub fn main() !void {
    var it = std.mem.tokenize(u8, input, "\n");

    var last: usize = std.math.maxInt(usize);
    var increments: usize = 0;
    while (it.next()) |line| {
        const n = try std.fmt.parseInt(usize, line, 10);
        if (n > last) increments += 1;
        last = n;
    }

    std.debug.print("result: {}\n", .{increments});
}

File: 2021/day01/part01/main.go
package main

import (
  "fmt"
  "os"
  "bufio"
  "log"
  "strconv"
)

func readFile(path string) []int {
  file, err := os.Open(path)
  if err != nil {
    log.Fatal(err)
  }
  defer file.Close()

  var nums []int
  scanner := bufio.NewScanner(file)
  for scanner.Scan() {
    txt := scanner.Text()
    if txt == "" { continue }
    num, err := strconv.Atoi(txt)
    if err != nil {
      log.Fatal(err)
    }
    nums = append(nums, num)
  }

  return nums
}

func main() {
  nums := readFile("input.txt")

  last := nums[0]
  increments := 0
  for _, n := range nums {
    if n > last { increments++ }
    last = n
  }

  fmt.Printf("result: %d\n", increments);
}
